sdk/keyvault/keyvault-keys/src/cryptographyClient.ts

```diff
@@ -152,35 +188,34 @@ export class CryptographyClient {
       | [EncryptParameters, EncryptOptions?]
       | [EncryptionAlgorithm, Uint8Array, EncryptOptions?]
   ): Promise<EncryptResult> {
+    this.ensureValid(await this.fetchKey(), KnownKeyOperations.Encrypt);
     const [parameters, options] = this.disambiguateEncryptArguments(args);
 
-    if (this.concreteClient.kind === "remote") {
-      return this.concreteClient.client.encrypt(parameters, options);
-    } else {
-      if (!isLocallySupported(parameters.algorithm)) {
-        throw new Error(
-          `Algorithm ${parameters.algorithm} is not supported for a local JsonWebKey.`
-        );
-      }
-      return this.concreteClient.client.encrypt(parameters, options);
+    const { span, updatedOptions } = this.createSpan("encrypt", options);
+
+    try {
+      const provider = await this.getProvider(KnownKeyOperations.Encrypt, parameters.algorithm);
+      return await provider.encrypt(parameters, updatedOptions);
+    } finally {
+      span.end();
     }
   }
 
   private disambiguateEncryptArguments(
     args: [EncryptParameters, EncryptOptions?] | [string, Uint8Array, EncryptOptions?]
-  ): [EncryptParameters, EncryptOptions?] {
+  ): [EncryptParameters, EncryptOptions] {
     if (typeof args[0] === "string") {
       // Sample shape: ["RSA1_5", buffer, options]
```

Comment:
does this shape sample continues to make sense?

Reply:
Yup, I think so but let me know if I missed anything!